File: Backend/EduPayAPI_Solution/Domain/Entities/Teacher.cs
namespace EduPayAPI.Domain.Entities;
public class Teacher
{
    public Guid Id { get; private set; }

    public string Code { get; private set; } = default!;
    public string Name { get; private set; } = default!;

    public string Email { get; private set; } = default!;
    public string? PhoneNumber { get; private set; }

    public TeacherStatus Status { get; private set; }

    

    public Guid BranchId { get; set; }
    public Guid SchoolYearId { get; set; }
    public Guid? DepartmentId { get; set; }

    // Navigation
    public Branch Branch { get; private set; } = default!;

    public SchoolYear SchoolYear { get; private set; } = default!;

    public Department? Department { get; private set; }

    protected Teacher() {}

     public Teacher(
        string code,
        string name,
        Guid branchId,
        Guid schoolYearId,
        string email,
        string? phoneNumber,
        Guid? departmentId = null)
    {
        Id = Guid.NewGuid();
        Code = code;
        Name = name;
        BranchId = branchId;
        SchoolYearId = schoolYearId;
        DepartmentId = departmentId;
        Email = email;
        PhoneNumber = phoneNumber;
        Status = TeacherStatus.Working;
    }

    public void Update (
        string code, 
        string name,
        string email,
        string? phoneNumber,
        TeacherStatus status
    )
    {
        Code = code;
        Name = name;
        Email = email;
        PhoneNumber = phoneNumber;
        Status = status;
    }
}

File: Frontend/edupay-app/src/lib/api/teacher.ts
// src/lib/api/teacher.ts
import { api } from "@/lib/axios";

export interface Teacher {
  id: string;
  code: string;
  name: string;
  email: string;
  phoneNumber: string | null;
  status: number;
  branchId: string;
  schoolYearId: string;
}

export interface GetTeachersParams {
  search?: string;
  status?: number;
  departmentId?: string;
  branchId: string;
  schoolYearId: string;
}

export interface CreateTeacherPayload {
  code: string;
  name: string;
  email?: string | null;
  phoneNumber?: string | null;
  branchId: string;
  schoolYearId: string;
}

export interface UpdateTeacherPayload {
  code: string;
  name: string;
  email: string;
  status: number;
  phoneNumber?: string | null;
}


// ================= GET LIST =================
export const getTeachers = async (
  params: GetTeachersParams
): Promise<Teacher[]> => {
  const res = await api.get("/teachers", {
    params,
  });

  return res.data.map((item: any) => ({
    id: item.id,
    code: item.code,
    name: item.name,
    email: item.email,
    phoneNumber: item.phoneNumber,
    status: item.status,
    branchId: item.branchId,
    schoolYearId: item.schoolYearId,
  }));
};

// ================= CREATE =================
export const createTeacher = async (payload: CreateTeacherPayload) => {
  const res = await api.post("/teachers", payload);
  return res; 
};

// ================= DELETE =================
export const deleteTeacher = async (id: string) => {
  const res = await api.delete(`/teachers/${id}`);
  return res;
};

export const updateTeacher = async (
  id: string,
  payload: UpdateTeacherPayload
) => {
  const res = await api.put(`/teachers/${id}`, payload);
  return res;
};